ci/build-gmt.sh

```diff
@@ -30,18 +34,19 @@ GMT_BUILD_TMPDIR=$(mktemp -d ${TMPDIR:-/tmp/}gmt.XXXXXX)
 cd ${GMT_BUILD_TMPDIR}
 
 # 2. Download GMT, GSHHG and DCW from GitHub
-git clone --depth 1 https://github.com/GenericMappingTools/gmt.git gmt
+curl -SLO https://github.com/GenericMappingTools/gmt/archive/${GMT_GIT_REF}.${EXT}
```

Comment:
> The only issue is that, the build script uses the tarballs provided by GitHub, which don't have any git history information. So the GMT version string doesn't contain commit hash and commit date. For example, building the master branch gives you 6.2.0 instead of 6.2_b234d76_2020.08.04

It would be better to have the commit hash. I haven't tried it before, but maybe try `git clone --single-branch --branch <branchname> <remote-repo>` to clone a single branch? See https://stackoverflow.com/questions/1911109/how-do-i-clone-a-specific-git-branch.

Reply:
OK, will see if a shallow clone is possible for a single branch or tag.

Reply:
Done. Now using:
```
git clone --depth=1 --single-branch --branch ${GMT_GIT_REF} https://github.com/GenericMappingTools/gmt
```
GMT_GIT_REF can be a branch name or a tag, NOT a commit hash (but most of the time we don't need to build a specific commit).